# Actividad Evaluable: Obtención de estadisticas descriptivas
#### Ethan Alejandro Alvarez Suarez - A00843413

In [78]:
#importa librerías
import pandas as pd
import matplotlib.pyplot as plt

In [79]:
diabetes = pd.read_csv("diabetes.csv")

# Descripción de Variables

- SkinThickness Skin Thickness (grasa local debajo de la piel - mm; 0 = NULL) : Cuantitativa Continua  
- Insulin Insulin (La cantidad de insulina que secreta el páncreas en respuesta a la glucosa - μU/mL (micro-units per milliliter); 0 = NULL) : Cuantitativa Continua  
- BMI Body Mass Index (Tamaño corporal total - kg/(m^2); 0 = NULL) : Cuantitativa Continua

In [80]:
#Rows & Columns
diabetes.iloc[:, 3:6].shape

(768, 3)

In [81]:
#Number of unique values for each dataset
diabetes.iloc[:, 3:6].nunique()

SkinThickness     51
Insulin          186
BMI              248
dtype: int64

# Variables Cuantitativas

### Medidas de tendencia central, STD, y Gráficos


## Exploración de Datos y Conclusiones

In [82]:
#number of null values in the datasets
diabetes[['SkinThickness', 'Insulin', 'BMI']].replace(0, pd.NA).isna().sum()

SkinThickness    227
Insulin          374
BMI               11
dtype: int64

In [83]:
mean_SkinThickness = diabetes['SkinThickness'].mean()
median_SkinThickness = diabetes['SkinThickness'].median()
mode_SkinThickness = diabetes['SkinThickness'].mode()
print("Mean_SkinThickness: ", mean_SkinThickness)
print("Median_SkinThickness: ", median_SkinThickness)
print("Mode_SkinThickness: ", mode_SkinThickness)

mean_Insulin = diabetes['Insulin'].mean()
median_Insulin = diabetes['Insulin'].median()
mode_Insulin = diabetes['Insulin'].mode()
print("Mean_Insulin: ", mean_Insulin)
print("Median_Insulin: ", median_Insulin)
print("Mode_Insulin: ", mode_Insulin)

mean_BMI = diabetes['BMI'].mean()
median_BMI = diabetes['BMI'].median()
mode_BMI = diabetes['BMI'].mode()
print("Mean_BMI: ", mean_BMI)
print("Median_BMI: ", median_BMI)
print("Mode_BMI: ", mode_BMI)

Mean_SkinThickness:  20.536458333333332
Median_SkinThickness:  23.0
Mode_SkinThickness:  0    0
Name: SkinThickness, dtype: int64
Mean_Insulin:  79.79947916666667
Median_Insulin:  30.5
Mode_Insulin:  0    0
Name: Insulin, dtype: int64
Mean_BMI:  31.992578124999998
Median_BMI:  32.0
Mode_BMI:  0    32.0
Name: BMI, dtype: float64


In [84]:
#min/max values; mean/median values of each table WITH NULL VALUES
diabetes.iloc[:, 3:6].describe()

SkinThickness     Insulin         BMI
count     768.000000  768.000000  768.000000
mean       20.536458   79.799479   31.992578
std        15.952218  115.244002    7.884160
min         0.000000    0.000000    0.000000
25%         0.000000    0.000000   27.300000
50%        23.000000   30.500000   32.000000
75%        32.000000  127.250000   36.600000
max        99.000000  846.000000   67.100000

In [85]:
#next lowest value
print(diabetes[['SkinThickness', 'Insulin', 'BMI']].replace(0, pd.NA).min())

SkinThickness       7
Insulin            14
BMI              18.2
dtype: object


In [86]:
#STD of table WITH NULL VALUES
diabetes[['SkinThickness', 'Insulin', 'BMI']].std()

SkinThickness     15.952218
Insulin          115.244002
BMI                7.884160
dtype: float64

### Data analysis excluding null values

In [87]:
#mean values of each table WITHOUT NULL VALUES
diabetes[['SkinThickness', 'Insulin', 'BMI']].replace(0, pd.NA).mean()

SkinThickness      29.15342
Insulin          155.548223
BMI               32.457464
dtype: object

In [88]:
#median values of each table WITHOUT NULL VALUES
diabetes[['SkinThickness', 'Insulin', 'BMI']].replace(0, pd.NA).median()

SkinThickness     29.0
Insulin          125.0
BMI               32.3
dtype: object

In [89]:
#mode values of each table WITHOUT NULL VALUES
diabetes[['SkinThickness', 'Insulin', 'BMI']].replace(0, pd.NA).mode()

SkinThickness Insulin   BMI
0            32     105  32.0

In [90]:
#STD of table WITHOUT NULL VALUES
diabetes[['SkinThickness', 'Insulin', 'BMI']].replace(0, pd.NA).std()

SkinThickness     10.476982
Insulin          118.775855
BMI                6.924988
dtype: object

# Conclusiónes:  

## Para el grosor de la piel:  
Rango: (0, 99)  
siguiente valor más bajo: 7  
numero de valores null: 22  
- datos CON valores NULL:  
promedio: 20  
valor medio: 23  
valor más común: 0  
STD: 16  
---  
- datos SIN valores NULL:
promedio: 29  
valor medio: 29  
valor más común: 32  
STD: 10  
---  
## Para la insulina:  
Rango: (0, 846)  
siguiente valor más bajo: 14  
numero de valores null: 374
- datos CON valores NULL:  
promedio: 79  
valor medio: 30.5  
valor más común: 0  
STD: 115  
---  
- datos SIN valores NULL:  
promedio: 125  
valor medio: 125  
valor más común: 105  
STD: 118  
---  
## Para el BMI:  
Rango: (0, 67.1)  
siguiente valor más bajo: 18.2  
numero de valores null: 11  
- datos CON valores NULL:  
promedio: 32.0  
valor medio: 32.0  
valor más común: 32.0  
STD: 7.9  
---  
- datos SIN valores NULL:  
promedio: 32.5
valor medio: 32.3  
valor más común: 32.0
STD: 6.9